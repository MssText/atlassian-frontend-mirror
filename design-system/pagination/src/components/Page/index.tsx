import React, { Component } from 'react';

import {
  Appearance,
  CustomThemeButton,
  CustomThemeButtonProps,
} from '@atlaskit/button';

type Diff<T, U> = T extends U ? never : T;

type PagePropsType = Diff<
  CustomThemeButtonProps,
  {
    appearance?: Appearance;
    autoFocus: boolean;
    isDisabled: boolean;
    isLoading: boolean;
    spacing: 'compact' | 'default' | 'none';
    shouldFitContainer: boolean;
    type: 'button' | 'submit';
  }
> & { page?: any };

export default class Page extends Component<PagePropsType> {
  render() {
    return <CustomThemeButton {...this.props} appearance="subtle" />;
  }
}
