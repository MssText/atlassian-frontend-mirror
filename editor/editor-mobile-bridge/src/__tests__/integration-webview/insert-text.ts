import { MobileTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-app-wrapper';
import {
  loadEditor,
  getADFContent,
  SELECTORS_WEB,
} from './_utils/afe-app-helpers';

MobileTestCase(
  'Insert Text: Foo Bar',
  // FIXME: ED-10399 Doesn't insert text into the DOM for iOS
  { skipPlatform: ['ios'] },
  async (client: any, testName: string) => {
    const page = await Page.create(client);
    await loadEditor(page);

    await page.insertText(SELECTORS_WEB.EDITOR, 'Foo Bar');
    const adfContent = await getADFContent(page);

    expect(adfContent).toMatchCustomDocSnapshot(
      `${testName} - ${page.platform()}.`,
    );
  },
);

MobileTestCase(
  'Insert Text: Hello World',
  // FIXME: ED-10399 Doesn't insert text into the DOM for iOS
  { skipPlatform: ['ios'] },
  async (client: any, testName: string) => {
    const page = await Page.create(client);
    await loadEditor(page);

    await page.insertText(SELECTORS_WEB.EDITOR, 'Hello World');
    const adfContent = await getADFContent(page);

    expect(adfContent).toMatchCustomDocSnapshot(
      `${testName} - ${page.platform()}.`,
    );
  },
);
