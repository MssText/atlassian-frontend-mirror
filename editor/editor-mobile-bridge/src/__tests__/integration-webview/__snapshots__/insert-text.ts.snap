// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insert Text: Foo Bar - Android. 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "Foo Bar",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Insert Text: Foo Bar - Android. 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "Foo Bar",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Insert Text: Hello World - Android. 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "Hello World",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Insert Text: Hello World - Android. 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "Hello World",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;
